awsconfig/providers/awsvault.go

```diff
@@ -0,0 +1,93 @@
+package providers
+
+import (
+	"bufio"
+	"context"
+	"errors"
+	"os/exec"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+type AWSVaultCredentialsProvider struct {
+	Profile string
+	Cache   *aws.CredentialsCache
+}
+
+func NewAWSVaultCredentialsProvider(profile string) *AWSVaultCredentialsProvider {
+	provider := &AWSVaultCredentialsProvider{
+		Profile: profile,
+	}
+
+	provider.Cache = aws.NewCredentialsCache(provider)
+
+	return provider
+}
+
+func (c AWSVaultCredentialsProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+	output, err := executeAwsVaultCommand(c.Profile)
+	if err != nil {
+		return aws.Credentials{}, errors.New("failed to execute aws-vault command: " + err.Error())
+	}
+
+	keyValueMap, err := parseKeyValuePairs(output)
+	if err != nil {
+		return aws.Credentials{}, errors.New("failed to parse aws-vault output: " + err.Error())
+	}
+
+	creds, err := mapToCredentials(keyValueMap)
+	if err != nil {
+		return aws.Credentials{}, errors.New("failed to map credentials: " + err.Error())
+	}
+
+	return creds, nil
+}
+
+func executeAwsVaultCommand(profile string) (string, error) {
+	cmd := exec.Command("aws-vault", "exec", profile, "--no-session", "--", "env")
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+
+	return string(output), nil
+}
```

Comment:
we should add better error messaging for if `aws-vault` was not resolvable. something like, "Credential provider was configured as `aws-vault`, but `aws-vault` was not installed or on your `$PATH`"